Update documentation about our documentation infrastructure at mystmd.org

choldgraf opened the pull request:
This updates our documentation infrastructure to make it more transparent for our team to understand how the docs at mystmd.org are hosted, how to update them, and how it all works.

I may have gotten some things factually wrong, so it would be great to get a review from somebody with more knowledge than me to make sure it is correct.

- Depends on https://github.com/jupyter-book/mystmd.org/pull/27 for the dispatch option

agoose77 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,47 @@ To build the MyST guide documentation:
 This will build the documentation locally so that you can preview what your changes will look like.
 
-## How this relates to the documentation at mystmd.org
+## How update the content at mystmd.org
```
```suggestion
## How to update the content at mystmd.org
```

agoose77 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,47 @@ To build the MyST guide documentation:
+
+When you change the content here or in most other MyST repositories, it will _not_ automatically update mystmd.org. You must take one extra step, described below.
+
+The content at mystmd.org is automatically updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.
```
@rowanc1 can you remind me if this is intentional that we don't deploy automatically? I could set up a web hook to trigger the deploy workflow.

choldgraf replied:
@agoose77 could we make that a follow-up issue? Or do you want to block this PR on that question?

agoose77 replied:
This is _not_ a blocker. I'm just looking for clarity on _why_ we have this policy. We should prioritise getting these docs merged, and revisit if needs be.

choldgraf replied:
I recommend:

- Creating an issue called `Make mystmd.org automatically update any time changes are merged into content repositories`
- Our first action in that issue should be to decide if we want to do it at all, or if something precludes us doing it.
- If we do want to do it, then the next step is to implement whatever webhooks you're talking about.
- If we don't want to do it, we write down a rationale for why not in that issue, and we close it as a record of the conversation+decision so that we don't have this uncertainty in the future

stefanv commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,47 @@ To build the MyST guide documentation:
+
+When you change the content here or in most other MyST repositories, it will _not_ automatically update mystmd.org. You must take one extra step, described below.
+
+The content at mystmd.org is automatically updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.
```
```suggestion
The content at mystmd.org is updated with [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.
```

Too confusing to use "automatically" here, since you just stated that the process is not automatic. This is a different automatic, I get it, but still confusing.

stefanv commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,47 @@ To build the MyST guide documentation:
+To trigger that action, you can either:
+
+1. **Trigger a workflow dispatch**. Go to [the Vercel action page](https://github.com/jupyter-book/mystmd.org/actions/workflows/deploy.yml). Click on *Run Workflow* and it will run.
+2. **Push a commit to `main`**. Any new commit to the `main` branch will trigger this action.
```
```suggestion
2. **Push a commit to `main`** (of [jupyter-book/mystmd.org](https://github.com/jupyter-book/mystmd.org)). Any new commit to the `main` branch will trigger this action.
```

To make clear, again, that it's not a commit to this repo that will do the trick.

stefanv commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,47 @@ To build the MyST guide documentation:
+1. **Trigger a workflow dispatch**. Go to [the Vercel action page](https://github.com/jupyter-book/mystmd.org/actions/workflows/deploy.yml). Click on *Run Workflow* and it will run.
+2. **Push a commit to `main`**. Any new commit to the `main` branch will trigger this action.
+
+In either case, once the action completes the content at `mystmd.org` will be updated.
```
A bit weird, because the "will be updated" part is already done. So, would consider removing the sentence, or changing to something like "Once the action completes, the content at `mystmd.org` should be up to date."

stefanv commented on docs/contribute-docs.md:
```diff
@@ -61,16 +82,11 @@ Here's a brief overview of where to look for things:
-2. **[dev facing]** if there were changes in `packages/*` we release the `@myst-theme/*` packages (changesets based ci step)
-3. **[dev facing]** we then run `make deploy-book` and `make deploy-article` which builds and bundles each theme, making commits to the https://github.com/myst-templates/book-theme and https://github.com/myst-templates/article-theme repos.
-4. **[user facing]** at that point those latest bundle commits will be what is pulled by `mystmd` clients - people who already have those downloaded need to `myst clean --all` (as we don't yet auto bump based on version changes, see [#854](https://github.com/jupyter-book/mystmd/issues/854) for updates).
+For example, `mystmd.org/guide` is pulled from [`jupyter-book/mystmd: /docs/`](https://github.com/jupyter-book/mystmd/tree/main/docs), and [`mystmd.org/jtex`](https://mystmd.org/jtex) is pulled from [`jupyter-book/mystmd: /jtex/docs/`](https://github.com/jupyter-book/mystmd/tree/main/packages/jtex/docs). It also includes some custom applications like the Sandbox and MyST demo on the landing page.
```
Good to have these year; a full list would be helpful, but annoying to duplicate. Could we link to the Vercel config, or commit that config to the repo like we do with Netlify?

stefanv replied:
I have never used Vercel, I'm afraid!

rowanc1 replied:
There is no config. Just select "remix" preset from the dropdown. We aren't doing anything fancy here!

stefanv reviewed (commented):
Thanks for these clarifications, Chris!

choldgraf commented:
Thanks all for the conversation and some clarification from @rowanc1 [in this discord thread](https://discord.com/channels/1083088970059096114/1177359316886503464/1361373261560287384).

I've updated this PR with a couple quick changes:

1. I now distinguish between "updating the theme" vs. "updating the live deployment at mystmd.org"
2. I clarified that the deployment to mystmd.org is controlled by a curvenote deployment that @rowanc1 manages, and so we need to ask him to update mystmd.org any time the content changes.
3. I added a cautionary note about updating the mystmd.org content if MyST has been updated

choldgraf commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,49 @@ To build the MyST guide documentation:
+
+The live website mystmd.org is **managed by a Curvenote deployment**. To update mystmd.org, you must make a deployment to Curvenote.
+
+Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. We are working on turning this process into a GitHub Action.
```
```suggestion
Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. See [this issue about standardizing team access to documentation infrastructure](https://github.com/jupyter-book/team-compass/issues/25) for making this easier for our team to use.
```

fperez commented:
Thanks @choldgraf for documenting this - the more "bog standard" and easy it is for anyone to replicate a deployment like the main MyST one, the better for gaining traction, users and contributors...

As I mentioned [in this issue](https://github.com/jupyter-book/team-compass/issues/25) I hope it can become "just a gh-pages deployment"  in the long run, but there may be reasons why that's not possible.

In any case, these doc updates look very useful.

Any reason not to merge and refine later, or are there any blocker issues in the PR as it is (I don't see any, but I'm not as deeply into this part of the project as others).

choldgraf commented:
I'm gonna merge this one so that we can continue iterating. It feels safe to try to me, and I think we should err on the side of merging and keeping momentum going forward

rowanc1 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,49 @@ To build the MyST guide documentation:
+### How to access the Vercel configuration for mystmd.org
+
+Currently the theme is run on **Rowan's** personal Vercel project.
+If you need to change the configuration reach out to him.
```
There is no config!

rowanc1 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,49 @@ To build the MyST guide documentation:
+
+The live website mystmd.org is **managed by a Curvenote deployment**. To update mystmd.org, you must make a deployment to Curvenote.
+
+Currently, the only way to update MyST's Curvenote deployment is to ask [@rowanc1](https://github.com/rowanc1) to do it. See [this issue about standardizing team access to documentation infrastructure](https://github.com/jupyter-book/team-compass/issues/25) for making this easier for our team to use.
```
Only content hosted on Curvenote, not the theme.

rowanc1 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,49 @@ To build the MyST guide documentation:
+
+When you change the content here or in most other MyST repositories, it will _not_ automatically update mystmd.org. This is because we use [a custom MyST theme](#about:mystmd.org) that aggregates content from several MyST repositories into a single website.
+
+The live website mystmd.org is **managed by a Curvenote deployment**. To update mystmd.org, you must make a deployment to Curvenote.
```
Only the content is hosted on Curvenote, the theme is not.

rowanc1 commented on docs/contribute-docs.md:
```diff
@@ -30,26 +30,49 @@ To build the MyST guide documentation:
+The [jupyter-book/mystmd.org](https://github.com/jupyter-book/mystmd.org) repository has a custom MyST theme that is used to control the content and structure of the site at mystmd.org. It is a [Remix](https://remix.run/) website and is deployed on [Vercel](https://vercel.com/). 
+
+(team-vercel-deployment)=
+### How to access the Vercel configuration for mystmd.org
```
There is no config, so not sure this section is actually useful?

rowanc1 commented on docs/contribute-docs.md:
```diff
@@ -61,16 +84,21 @@ Here's a brief overview of where to look for things:
+### How do I deploy changes to the mystmd.org theme?
+
+After updating the theme at `mystmd.org`, we must [deploy it to Vercel](#team-vercel-deployment) in order to make it available for use.
+We use [a GitHub Action to deploy to our Vercel project](https://github.com/jupyter-book/mystmd.org/blob/main/.github/workflows/deploy.yml). When that action is run, it will pull the latest docs content from several MyST repositories, and push them to our Vercel deployment.
```
The website doesn't need to be redeployed to get new content live.

rowanc1 reviewed (changes requested):
A few incorrect things here @choldgraf

choldgraf commented:
Ok I'll update with a follow up PR when I have a moment
